### DISCUSSION TITLE
Allow querying by root owner

### DISCUSSION THREAD

**Discussion by shifuma** [craftcms/cms#15790] [Ideas]
TLDR: There are times when it would be nice to get deeply nested entries, without needing to go through all levels: `craft.entries().field('nestedMatrixField').rootOwner(entry).all()`

---

Some more information on my setup:

- Events have various relation and other fields (category, date, day of the week etc)
- Companies - a Matrix field on Events
- Presentations - a Matrix field on Companies

On a company's page, I am listing all of their presentations. However, I want to then filter those based on fields on Events. For example, from Company A's page, I want to see all of their presentations made at Events on a Saturday. That's where I thought this would be perfect.

```
-- `companies/_entry.twg` --
{% set events = craft.entries().section('events').dayOfTheWeek('Saturday').ids()
{% set presentations = craft.entries().field('presentations').rootOwnerId(events).all() %}
```

**Comment by shifuma** [comment]
Related: https://github.com/craftcms/cms/discussions/15534

**Comment by rungta** [comment]
+1 for this feature. Example use case: we have a data model for storing a book’s table of contents, where a book contains multiple chapters and these chapters are grouped in sub-sections. The sub-sections can also be grouped together inside a parent sub-section, infinitely. So we end up with a structure such as:

Book One (root entry):
- Contents (matrix field)
   - Sub-section A (matrix field)
      - Chapter A.1
      - Chapter A.2
   - Sub-section B (matrix field)
      - Sub-section B.1 (matrix field)
         - Chapter B.1.1
         - Chapter B.1.1
      - Sub-section B.2 (matrix field)
        - Sub-section B.2.1 (matrix field)
        - Sub-section B.2.2 (matrix field)
           - Chapter B.2.2.1

Note that _Books_ and _Chapters_ are both separate sections.

In order to query all chapters of a book, would be great to write `craft.entries.type('chapters').rootOwner(bookEntry)`
